src/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -896,5 +966,153 @@ public static void InvalidDateTimeStyles()
             DateTimeStyles style = DateTimeStyles.AssumeLocal | DateTimeStyles.AssumeUniversal;
             AssertExtensions.Throws<ArgumentException>("style", () => DateTime.ParseExact(strDateTime, formats, provider, style));
         }
+
+        public static IEnumerable<object[]> Parse_ValidInput_Suceeds_MemberData()
+        {
+            yield return new object[] { "1234 12", null, new DateTime(1234, 12, 1, 0, 0, 0) };
+            yield return new object[] { "12 1234", null, new DateTime(1234, 12, 1, 0, 0, 0) };
+            yield return new object[] { "12 1234 11", null, new DateTime(1234, 12, 11, 0, 0, 0) };
+            yield return new object[] { "1234 12 13", null, new DateTime(1234, 12, 13, 0, 0, 0) };
+            yield return new object[] { "12 13 1234", null, new DateTime(1234, 12, 13, 0, 0, 0) };
+            yield return new object[] { "1 1 1", null, new DateTime(2001, 1, 1, 0, 0, 0) };
+            yield return new object[] { "2 2 2Z", null, null };
+            yield return new object[] { "#10/10/2095#\0", null, new DateTime(2095, 10, 10, 0, 0, 0) };
```

Comment:
This can easily fail if the user has some non English cultures. something like ar-SA. I would suggest to add culture objects to the data